js/main.js

```diff
@@ -7,23 +7,37 @@ var entries = document.querySelector('ul');
 var entriesAnchor = document.querySelector('#entries-anchor');
 var newButton = document.querySelector('.new-button');
 var views = document.querySelectorAll('.view');
+var newEditEntry = document.querySelector('.new-edit-entry');
+var edit = false;
 
 switchViews(data.view);
 
 function saveEntry(event) {
   event.preventDefault();
-  var entry = {};
-  entry.title = form.elements.title.value;
-  entry.photoUrl = form.elements['photo-url'].value;
-  entry.notes = form.elements.notes.value;
-  entry.id = data.nextEntryId;
-  data.nextEntryId++;
-  data.entries.unshift(entry);
-  img.setAttribute('src', 'images/placeholder-image-square.jpg');
-  entries.prepend(loadEntry(entry));
+  if (edit === false) {
+    var entry = {};
+    entry.title = form.elements.title.value;
+    entry.photoUrl = form.elements['photo-url'].value;
+    entry.notes = form.elements.notes.value;
+    entry.id = data.nextEntryId;
+    data.nextEntryId++;
+    data.entries.unshift(entry);
+    img.setAttribute('src', 'images/placeholder-image-square.jpg');
+    entries.prepend(loadEntry(entry));
+  } else {
+    data.editing.title = form.elements.title.value;
+    data.editing.photoUrl = form.elements['photo-url'].value;
+    data.editing.notes = form.elements.notes.value;
+    edit = false;
+    data.editing = null;
+  }
   form.reset();
+  img.setAttribute('src', 'images/placeholder-image-square.jpg');
+  entries.innerHTML = '';
+  appendEntries();
   switchViews('entries');
-  window.scroll(0, 0);
+  newEditEntry.textContent = 'New Entry';
+  entriesAnchor.className = 'margin-left-1rem';
```

Comment:
Why is this line necessary?

Reply:
Still not sure why this line is necessary 🤔 

Does the anchor tag need to look different on different views?

Reply:
Oops, I was under the (false) impression that we had to hide the anchor element in edit mode.

Reply:
Oh oops I thought we had to hide the anchor element in edit mode. 😅